service-base/src/test/java/org/eclipse/hono/service/auth/device/UsernamePasswordCredentialsTest.java

```diff
@@ -122,13 +116,30 @@ public void testMatchesCredentialsSucceedsForMatchingPassword() throws NoSuchAlg
         assertTrue(credentials.matchesCredentials(candidateSecret));
     }
 
+    /**
+     * Verifies that credentials support Bcrypt hash function.
+     */
+    @Test
+    public void testMatchesCredentialsSucceedsForMatchingBCryptPassword() {
+
+        // GIVEN a secret on record that uses bcrypt as the hash function
+        final JsonObject candidateSecret = CredentialsObject.emptySecret(null, null);
+        candidateSecret.put(CredentialsConstants.FIELD_SECRETS_PWD_HASH, "$2a$12$BjLeC/gqcnEyk.XNo2qorul.a/v4HDuOUlfmojdSZXRSFTjymPdVm");
```

Comment:
Why don't you use `CredentialsObject.hashedPasswordSecret()` for creating the candidate secret?

Reply:
I wanted to test it against default hash stored in Kapua, just to avoid any surprises later on.